pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/properties/PreserveStashesJobPropertyLoadTest.java

```diff
@@ -0,0 +1,113 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.modeldefinition.properties;
+
+import org.jenkinsci.plugins.pipeline.modeldefinition.AbstractDeclarativeTest;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.flow.StashManager;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.util.concurrent.Callable;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class PreserveStashesJobPropertyLoadTest extends AbstractDeclarativeTest {
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    @Rule
+    public LoggerRule logger = new LoggerRule();
+
+    @Issue("JENKINS-45455")
+    @Test
+    public void stashWithLoadCount() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "stashWithLoadCount");
+                p.setDefinition(new CpsFlowDefinition(pipelineSourceFromResources("properties/stashWithLoadCount"), true));
+
+                // Build 10 times to make sure get 10 total builds.
+                for (int i = 0; i < 10; i++) {
+                    story.j.waitForCompletion(p.scheduleBuild2(0).waitForStart());
+                }
+            }
+        });
+
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob p = (WorkflowJob)story.j.jenkins.getItem("stashWithLoadCount");
+                assertNotNull(p);
+                PipelineRunLoadCounter.prepare(p);
+
+                // PreserveStashesJobProperty.StashClearingListener and .SaveStashes shouldn't trigger a full load.
+                assertEquals(0, PipelineRunLoadCounter.countLoads(p, new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            story.j.buildAndAssertSuccess(p);
+                        } catch (Exception e) {
+                            // This will never happen.
```

Comment:
orly? how about an `assert`, just in case

Reply:
Sure thing.